launchers/junit/src/testFixtures/java/org/eclipse/dataspaceconnector/junit/launcher/EdcRuntimeExtension.java

```diff
@@ -0,0 +1,175 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.junit.launcher;
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+import java.util.stream.Stream;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * A JUnit extension for running an embedded EDC runtime as part of a test fixture.
+ * The runtime obtains a classpath determined by the Gradle build.
+ * <p>
+ * This extension attaches a EDC runtime to the {@link BeforeTestExecutionCallback} and {@link AfterTestExecutionCallback} lifecycle hooks. Parameter injection of runtime services is supported.
+ */
+public class EdcRuntimeExtension extends EdcExtension {
+    final String moduleName;
+    final Map<String, String> properties;
+    private Thread runtimeThread;
+    private static final String GRADLE_WRAPPER_UNIX = "gradlew";
+    private static final String GRADLE_WRAPPER_WINDOWS = "gradlew.bat";
+    private static final String GRADLE_WRAPPER;
+
+    static {
+        GRADLE_WRAPPER = (System.getProperty("os.name").toLowerCase().contains("win")) ? GRADLE_WRAPPER_WINDOWS : GRADLE_WRAPPER_UNIX;
+    }
+
+    public EdcRuntimeExtension(String moduleName, Map<String, String> properties) {
+        this.moduleName = moduleName;
+        this.properties = Map.copyOf(properties);
+    }
+
+    @Override
+    public void beforeTestExecution(ExtensionContext extensionContext) throws Exception {
+
```

Comment:
Consider adding more logging here to have more clarity what modules are being started. Would be nice to see something like:
```
Starting "consumer" extension
....
Extension "consumer" running

Starting "provider" extension"
...

Reply:
Updated.